pkg/controllers/management/clusterstats/statsaggregator.go

```diff
@@ -160,8 +165,31 @@ func (s *StatsAggregator) aggregate(cluster *v3.Cluster, clusterName string) err
 		v32.ClusterConditionNoMemoryPressure.False(cluster)
 	}
 
+	var oldVersion int
+	if cluster.Status.Version != nil {
+		oldVersion, err = strconv.Atoi(cluster.Status.Version.Minor)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Can this be moved inside the if-statement so it only happens when necessary?

Reply:
Do you mean the other if statement on line 180? It has to happen before `s.updateVersion` is called because that is when `cluster.Status.Version` changes, so this is the only place to put it.